### PR TITLE
Don't animate the creation of new views

### LINKED ISSUE #202: Wrong bounds animation in 0.12
Did anything change in version `0.12` to animate bounds change between states?

I'm overriding `boundsAnimationFromPreviousComponent:` but the frame from which the animation starts in `0.12` is wrong.

From the diff I can see that a new function `CKComponentBoundsAnimationApply` has been added. It makes me wonder whether there was a regression perhaps?

### PR DESCRIPTION
**Pull request by adamjernst**
Also disables animation when recycling a view to show a logically distinct component (as determined by `scopeFrameToken`).

Fixes #202

### CODE REVIEW

File: ComponentKit/Core/CKComponent.mm [hunk 1]
```diff
@@ -125,6 +126,7 @@ - (CKComponentViewContext)viewContext
 
   UIView *v = effectiveContext.viewManager->viewForConfiguration([self class], viewConfiguration);
   if (v) {
+    CKMountAnimationGuard g(v.ck_component, self);
```
Comment (modocache) [inline ComponentKit/Core/CKComponent.mm]: Wow, this is super cool. C++ is neat.

File: ComponentKitApplicationTests/CKComponentBoundsAnimationTests.mm [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  NSSet *secondMountedComponents = CKMountComponentLayout(secondLayout, container);
+
+  CKBoundsAnimationRecordingView *v = (CKBoundsAnimationRecordingView *)secondResult.component.viewContext.view;
+  XCTAssertFalse(v.animatedLastBoundsChange);
+
+  [firstMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+  [secondMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+@end
```
Comment (modocache) [inline ComponentKitApplicationTests/CKComponentBoundsAnimationTests.mm]: Are there no tests that actions are re-enabled once the mount animation guard goes out of scope? Should there be? Disabling the actions affects the global state, right?

Comment (adamjernst) [inline ComponentKitApplicationTests/CKComponentBoundsAnimationTests.mm]: True. Interesting! Let me consider.

**Comment by modocache** [comment]
A few questions but LGTM! :+1:

**Comment by adamjernst** [comment]
Follow-up test coverage for that sounds like a great idea.

**Pull request merged**